File: Algorithms/C++/Easy/6_ZigZag_Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (s.empty() || numRows <= 1) {
            return s;
        }

        string result;
        vector<string> rows(numRows, "");

        int cycle = numRows + numRows - 2;
        for (int i = 0; i < s.length(); i++) {
            int row = i % cycle;
            if (row < numRows) {
                rows[row] += s[i];
            } else {
                rows[cycle - row] += s[i];
            }
        }

        for (int i = 0; i < numRows; i++) {
            result += rows[i];
        }

        return result;
    }
};

File: Algorithms/Swift/Easy/6_ZigZag_Conversion.swift
class Solution {
    func convert(s: String, _ numRows: Int) -> String {
        if s.isEmpty || numRows <= 1 {
            return s
        }

        var result = ""
        var rows = [String](count: numRows, repeatedValue: "")

        let cycle = numRows + numRows - 2
        for (index, character) in s.characters.enumerate() {
            let rowIndex = index % cycle
            if rowIndex < numRows {
                rows[rowIndex] += String(character)
            } else {
                rows[cycle - rowIndex] += String(character)
            }
        }

        for row in rows {
            result += row
        }

        return result
    }
}
